src/open_inwoner/search/views.py

```diff
@@ -64,13 +69,33 @@ def search(self, form):
 
         # Check if the query exactly matches with a case that belongs to the user
         if search_params := get_user_fetch_parameters(self.request):
-            if client := build_zaken_client():
-                cases = client.fetch_cases(**search_params, identificatie=query)
-                if cases and len(cases) == 1:
+            clients = build_zaken_clients()
+            proxy = MultiZgwClientProxy(clients)
+            case_results = proxy.fetch_cases(
+                **search_params,
+                identificatie=query,
+            )
+            if case_results.has_errors:
+                self.log_system_action("unable to retrieve cases", user=user)
+
+            # TODO: We should simply return multiple cases in the search results,
+            # rather than redirect. For now, we maintain the existing behavior
+            # by returning and redirect to the first case found, if any.
+            if len(case_results.join_results()) > 1:
+                logger.error("found multiple cases for a single set of search params")
+
+            for case_result in case_results.successful_responses:
```

Comment:
See my earlier comment about naming the return value of `proxy`. Could and should be streamlined to make clear that the proxy does one thing: return a list of responses retrieved by some underlying client(s).

So here perhaps (starting from L. 74):
```
proxy_result = proxy.fetch_cases(
                **search_params,
                identificatie=query,
            )

...code...

for response in proxy_result.successful_responses:
    ...
```